### PR TITLE
1237 update deadletter status

### PR DESCRIPTION
**Pull request by ian-j-abbott-accenture**
## Description

- Update logic between getting the dataset status

## Reminders:

- Be mindful of impacts of changing Major/Minor/Patch versions of each elixir app:
  - [ ] If updating patch version, are you sure there are no chart changes required to maintain functionality? If so, you should bump minor version instead.
  - [ ] If updating Major or Minor versions , did you update the sauron chart configuration?
- [ ] If changing elixir code in an "app", did you update the relevant version
      in `mix.exs`?
- [ ] If altering an API endpoint, was the relevant postman collection updated?
  - [ ] If a new version of `smart_city` is being used (new fields on a struct), were the relevant postman collections updated?

### CODE REVIEW

File: apps/forklift/lib/forklift/data_writer.ex [hunk 1]
```diff
@@ -218,4 +223,18 @@ defmodule Forklift.DataWriter do
       dataset_id: dataset_id
     }
   end
+
+  defp create_ingestion_complete_data(dataset_id, ingestion_id, actual_message_count, extract_time) do
+    {expected_message_count, _} =
+      Redix.command!(:redix, ["GET", ingestion_id])
```
Comment (RyanRConaway) [inline apps/forklift/lib/forklift/data_writer.ex:229]: I think this Redis key should be ingestion_id + extraction start time.

Just using the ingestion ID potentially leads to race conditions between ingestion triggers. If another ingestion starts before this one is complete for some reason, the expected message count would have a race condition.

This is more of a consideration if push-based ingestions were ever added (Which is likely a feature if UrbanOS ever handled IoT data). An ingestion could have 2 concurrent ingestions running.

File: apps/reaper/lib/reaper/data_extract/processor.ex [hunk 2]
```diff
@@ -182,4 +184,14 @@ defmodule Reaper.DataExtract.Processor do
       dataset_id: dataset_id
     }
   end
+
+  defp create_data_retrieved_count_message(dataset_id, ingestion_id, expected_message_count, extraction_start_time) do
```
Comment (RyanRConaway) [inline apps/reaper/lib/reaper/data_extract/processor.ex:188]: Is this function used?

File: apps/reaper/lib/reaper/data_extract/processor.ex [hunk 3]
```diff
@@ -64,14 +64,16 @@ defmodule Reaper.DataExtract.Processor do
 
     wait_for_completion([producer_stage, validation_stage, schema_stage, load_stage])
 
+    messages_processed_count = messages_processed_count(cache_name)
```
Comment (RyanRConaway) [inline apps/reaper/lib/reaper/data_extract/processor.ex:67]: Why rearrange the messages_processed_count?

File: apps/reaper/mix.exs [hunk 4]
```diff
@@ -4,7 +4,7 @@ defmodule Reaper.MixProject do
   def project do
     [
       app: :reaper,
-      version: "2.0.42",
+      version: "2.0.43",
```
Comment (RyanRConaway) [inline apps/reaper/mix.exs:7]: Based on the changes I'm seeing. Looks like reaper may not need to be changed at all?

File: apps/reaper/lib/reaper/data_extract/processor.ex (hunk 3 above)
Comment (ian-j-abbott-accenture) [reply apps/reaper/lib/reaper/data_extract/processor.ex:67]: Another symptom of re-doing things a few times and forgetting what all I did...

**Pull request merged**